wrapper/templates/_wrapper.pyx

```diff
@@ -468,3 +472,71 @@ def save_fortran_restart():
 def cleanup():
     """Call the Fortran cleanup routines, which clear memory and write final restart files."""
     cleanup_subroutine()
+
+
+DiagnosticInfo = namedtuple(
+        'DiagnosticInfo', ['axes', 'module_name', 'name', 'description', 'unit'])
+
+
+cdef _get_diagnostic_info_by_index(int i):
+    cdef int ax
+    cdef int axes[1]
+    cdef char name[128]
+    cdef char mod_name[128]
+    cdef char desc[128]
+    cdef char unit[128]
+
+    get_metadata_diagnostics(&i, axes, &mod_name[0], &name[0], &desc[0], &unit[0])
```

Comment:
I know this is cut from the other repo but curious why `axes` is passed by value (I think) and the rest of the args are passed by reference?